back/app/services/permissions/user_requirements_service.rb

```diff
@@ -55,10 +51,17 @@ def base_requirements(permission)
         group_membership: 'dont_ask'
       }
     }
+  end
+
+  def base_requirements(permission)
+    everyone = base_requirements_template(permission).deep_dup.tap do |requirements|
+      # Custom fields and group membership can now be required on all permissions
+      requirements[:custom_fields] = requirements_custom_fields(permission).to_h { |field| [field.key, (field.required ? 'require' : 'ask')] }
+      requirements[:special][:group_membership] = 'require' if @check_groups && permission.groups.any?
+    end
```

Comment:
Note: Fields and group membership now available for all permitted_by values - well, in theory as the front-end does not yet support configuring it